macros/src/lib.rs

```diff
@@ -149,18 +211,76 @@ pub fn interrupt(
     };
     let vector = format!("__vector_{}", vect);
     let vector_ident = syn::Ident::new(&vector, proc_macro2::Span::call_site());
+    let vector_ident_s = vector_ident.to_string();
 
     quote::quote! (
-        #[no_mangle]
-        pub unsafe extern "avr-interrupt" fn #vector_ident() {
-            #ident();
+        #[doc(hidden)]
+        #[export_name = #vector_ident_s]
+        pub unsafe extern "avr-interrupt" fn #tramp_ident() {
+            #ident(
+                #(#resource_args),*
+            )
         }
 
-        #(#attrs)*
-        #[allow(non_snake_case)]
-        #unsafety fn #ident() {
-            #(#stmts)*
-        }
+        #f
```

Comment:
Same here, I think the vector function should be annotated with `#[doc(hidden)]`.